k8s/upgrades/0.9.0-1.0.0/cstor_pool_upgrade.sh

```diff
@@ -0,0 +1,279 @@
+pending_pods=$(kubectl get po -n $ns \
+    -l app=cstor-pool,openebs.io/storage-pool-claim=$spc \
+    -o jsonpath='{.items[?(@.status.phase!="Running")].metadata.name}')
+rc=$?; if [ $rc -ne 0 ]; then echo "Failed to get pool pods related to spc: $spc | Exit Code: $rc"; error_msg; exit 1; fi
+
+
+## If any deployments pods are in not running state then exit the upgrade process ###
+if [ $(echo $pending_pods | wc -w) -ne 0 ]; then
+    echo "To continue with upgrade script make sure all the pool deployment pods corresponding to $spc must be in running state"
+    error_msg
+    exit 1
+fi
+
+patch_blockdevice_list_for_spc $spc
+
+### Get the csp list which are related to the given spc ###
+csp_list=$(get_csp_list $spc)
+
+#### Get required info from current csp and use the info while upgrading ####
+for csp_name in `echo $csp_list | tr ":" " "`; do
+    ## Check CSP version ##
+    version=$(verify_openebs_version "csp" $csp_name)
+    rc=$?
+    if [ $rc -ne 0 ]; then
+        exit 1
+    elif [ $version == $pool_upgrade_version ]; then
+        continue
+    fi
+
+    ## Get disk info from corresponding sp ##
+    sp_name=$(kubectl get sp \
+              -l openebs.io/cstor-pool=$csp_name \
+              -o jsonpath="{.items[*].metadata.name}")
+    rc=$?; if [ $rc -ne 0 ]; then echo "Failed to get sp related to csp: $csp_name | Exit Code: $rc"; error_msg; exit 1; fi
+
+    ## Below command will give the output in form of
+    ## [disk-1 disk-2 disk-3 disk-4]
+    sp_disk_list=$(kubectl get sp $sp_name \
+                   -o jsonpath="{.spec.disks.diskList}" | \
+                    tr "[]" " ")
+    rc=$?; if [ $rc -ne 0 ]; then echo "Failed to get disks related to sp: $sp_name | Exit Code: $rc"; error_msg; exit 1; fi
+
+    ## Below snippet will get related info regarding block device and format
+    ## information in below format and save it to corresponding csp name
+    ## "group": [
+    ##    {
+    ##         "blockDevice": [
+    ##             {
+    ##                 "deviceID": "/var/openebs/sparse/6-ndm-sparse.img",
+    ##                 "inUseByPool": true,
+    ##                 "name": "sparse-177b6bc2ae2dd332c7a384a02179368b"
+    ##             },
+    ##             {
+    ##                 "deviceID": "/var/openebs/sparse/9-ndm-sparse.img",
+    ##                 "inUseByPool": true,
+    ##                 "name": "sparse-2239d60eb46b3c26b0428fae4d15c88a"
+    ##             }
+    ##         ]
+    ##    }
+    ## ]
+    csp_disk_list=""
+    for disk_name in $sp_disk_list; do
+         ## Assuming device Id present in first index
+         device_id=$(kubectl get disk $disk_name -o jsonpath="{.spec.devlinks[0].links[0]}")
+         if [ -z $device_id ]; then
+             device_id=$(kubectl get disk $disk_name -o jsonpath="{.spec.path}")
+         fi
+         csp_disk_list=$(make_csp_disk_list "$csp_disk_list" "$device_id" "$disk_name")
```

Comment:
hope passing double quotes '"' is fine here

Reply:
Yes It will treat things present in double quote treat as single argument